core-bundle/contao/drivers/DC_Folder.php

```diff
@@ -2294,7 +2294,7 @@ protected function save($varValue)
 			// Check the full path to see if the file extension has changed, because if
 			// $this->strExtension is empty, a new extension could otherwise be added to
 			// $varValue and change the file type!
-			if (Path::getExtension($varValue . $this->strExtension) !== Path::getExtension($this->varValue . $this->strExtension))
+			if (!is_dir(Path::join($this->strRootDir, $this->strPath, $this->varValue, $this->strExtension)) && Path::getExtension($varValue . $this->strExtension) !== Path::getExtension($this->varValue . $this->strExtension))
```

Comment:
```suggestion
			if ($this->strExtension && Path::getExtension($varValue . $this->strExtension) !== Path::getExtension($this->varValue . $this->strExtension))
```

`$this->strExtension` is empty if it is a folder, so this simple check (without `is_dir()`) should suffice, shouldn't it?

Reply:
No, I think that would re-introduce the security issue. With this check you could again rename a file from `foo` to `foo.exe`.

Reply:
Of course 🤦‍♂️